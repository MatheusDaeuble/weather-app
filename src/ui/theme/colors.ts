export enum Colors {
  // PRIMARY
  PRIMARY = '#2E89DF',

  // SECONDARY
  SECONDARY = '#809FFF',

  WHITE = '#FFFFFF',
  BLACK = '#000000',

  // FEEDBACK COLORS
  SUCCESS = '#0E934C',
  WARNING = '#FF8B2C',
  DANGER = '#EB5757',

  // GRAY SCALE
  GRAY_DARK = '#4B4F58',
  GRAY = '#949494',
  GRAY_LIGHT = '#CECECE',
}

export type Color =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black'
  | 'success'
  | 'warning'
  | 'danger'
  | 'gray-dark'
  | 'gray'
  | 'gray-light';

export const colors: { [key in Color]: string } = {
  'primary': Colors.PRIMARY,
  'secondary': Colors.SECONDARY,
  'white': Colors.WHITE,
  'black': Colors.BLACK,
  'success': Colors.SUCCESS,
  'danger': Colors.DANGER,
  'warning': Colors.WARNING,
  'gray-dark': Colors.GRAY_DARK,
  'gray': Colors.GRAY,
  'gray-light': Colors.GRAY_LIGHT,
};

export const defaultTheme = {
  colors,
};
