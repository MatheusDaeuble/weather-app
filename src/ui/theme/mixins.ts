import { Dimensions, PixelRatio } from 'react-native';

const { width: WINDOW_WIDTH, height: WINDOW_HEIGHT } = Dimensions.get('window');
const guidelineBaseWidth = 375;
const guidelineBaseHeight = 700;

export const scaleSize = (size: number) =>
  Number(((WINDOW_WIDTH / guidelineBaseWidth) * size).toFixed(0));

export const scaleHeight = (size: number) =>
  Number(((WINDOW_HEIGHT / guidelineBaseHeight) * size).toFixed(0));

export const scaleFont = (size: number) =>
  Number((size * PixelRatio.getFontScale()).toFixed(0));
