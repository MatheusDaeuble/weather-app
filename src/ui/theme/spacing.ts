import { scaleSize } from './mixins';

export const SCALE_28 = scaleSize(28);
export const SCALE_24 = scaleSize(24);
export const SCALE_20 = scaleSize(20);
export const SCALE_16 = scaleSize(16);
export const SCALE_12 = scaleSize(12);
export const SCALE_8 = scaleSize(8);
export const SCALE_4 = scaleSize(4);
