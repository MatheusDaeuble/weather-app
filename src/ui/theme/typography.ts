import { scaleFont } from './mixins';

// FONT FAMILY
export const FONT_FAMILY_REGULAR = 'OpenSans_400Regular';
export const FONT_FAMILY_BOLD = 'OpenSans_700Bold';

// FONT SIZE
export const FONT_SIZE_64 = scaleFont(62);
export const FONT_SIZE_26 = scaleFont(26);
export const FONT_SIZE_18 = scaleFont(18);
export const FONT_SIZE_16 = scaleFont(16);
export const FONT_SIZE_14 = scaleFont(14);

// LINE HEIGHT
export const LINE_HEIGHT_93 = scaleFont(93);
export const LINE_HEIGHT_39 = scaleFont(39);
export const LINE_HEIGHT_27 = scaleFont(27);
export const LINE_HEIGHT_24 = scaleFont(24);
export const LINE_HEIGHT_21 = scaleFont(21);
