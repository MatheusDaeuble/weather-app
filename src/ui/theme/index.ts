import * as Colors from './colors';
import * as Spacing from './spacing';
import * as TypographyAttributes from './typography';
import * as Mixins from './mixins';

export { TypographyAttributes, Spacing, Colors, Mixins };
