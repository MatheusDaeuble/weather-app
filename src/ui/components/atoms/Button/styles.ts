import styled from 'styled-components/native';
import { ButtonProps } from '.';
import Typography from '../Typography';
import { Spacing } from 'ui/theme';
import { colors } from 'ui/theme/colors';

export const BUTTON_HEIGHT = 54;

type ContainerProps = Omit<ButtonProps, 'label'>;

export const Container = styled.TouchableOpacity.attrs({
  activeOpacity: 0.4,
})<ContainerProps>`
  width: 100%;
  padding: ${Spacing.SCALE_12}px;
  min-height: ${BUTTON_HEIGHT}px;
  background-color: ${({ color, disabled }) =>
    colors[disabled ? 'gray-light' : color]};
  justify-content: center;
  align-items: center;
  border-radius: ${Spacing.SCALE_8}px;
`;

export const Label = styled(Typography).attrs({
  variant: 'h3',
  align: 'center',
})``;
