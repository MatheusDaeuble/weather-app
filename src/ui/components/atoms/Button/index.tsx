import React from 'react';
import { ActivityIndicator } from 'react-native';
import { Color, colors } from 'ui/theme/colors';
import { Container, Label } from './styles';

export interface ButtonProps {
  label: string;
  color?: Color;
  onPress?: () => void;
  labelColor?: Color;
  disabled?: boolean;
  loading?: boolean;
}

const Button = ({
  label,
  color = 'primary',
  disabled = false,
  loading = false,
  onPress = () => {},
  ...props
}: ButtonProps) => {
  return (
    <Container
      onPress={onPress}
      disabled={disabled || loading}
      color={color}
      {...props}
    >
      {loading ? (
        <ActivityIndicator size='small' color={colors.white} />
      ) : (
        <Label color='white'>{label}</Label>
      )}
    </Container>
  );
};

export default Button;
