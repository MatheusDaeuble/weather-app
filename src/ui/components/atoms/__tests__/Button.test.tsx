import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { Color, colors } from 'ui/theme/colors';
import { View } from 'react-native';
import Button from '../Button';

describe('Button Atom', () => {
  describe('Render color', () => {
    Object.keys(colors).forEach((color: Color) => {
      it(`Render color: ${color}`, () => {
        const component = renderer
          .create(<Button color={color} label='Testing' />)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });

    it('Render disabled', () => {
      const component = renderer
        .create(<Button disabled label='Testing' />)
        .toJSON();
      expect(component).toMatchSnapshot();
    });

    it('Render in loading', () => {
      const component = renderer
        .create(<Button loading label='Testing' />)
        .toJSON();
      expect(component).toMatchSnapshot();
    });

    it('should be able disabled while loading', () => {
      const { getByTestId } = render(
        <View testID='testing'>
          <Button loading label='Testing' />
        </View>
      );

      expect(
        getByTestId('testing').findByProps({ disabled: true })
      ).toBeTruthy();
    });
  });
});
