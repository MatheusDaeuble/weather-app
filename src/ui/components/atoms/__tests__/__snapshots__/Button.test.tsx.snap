// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom Render color Render color: black 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: danger 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EB5757",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: gray 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#949494",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: gray-dark 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4B4F58",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: gray-light 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#CECECE",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: primary 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2E89DF",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: secondary 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#809FFF",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: success 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0E934C",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: warning 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF8B2C",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render color: white 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#CECECE",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    align="center"
    color="white"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 36,
          "lineHeight": 54,
          "textAlign": "center",
        },
        Object {},
      ]
    }
    variant="h3"
  >
    Testing
  </Text>
</View>
`;

exports[`Button Atom Render color Render in loading 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#CECECE",
      "borderRadius": 16,
      "justifyContent": "center",
      "minHeight": 54,
      "opacity": 1,
      "paddingBottom": 24,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    color="#FFFFFF"
    size="small"
  />
</View>
`;
